# Description: fix compilation with gcc-4.5 (and corresponding libstdc++)
#  Some deprecated typedefs are finaly removed
#  Constructors need not be explicitly named
# Author: George Danchev <danchev@debian.org>
# Forwarded: no
# Debian-Bug: #564991
--- a/src/dictgui.cpp
+++ b/src/dictgui.cpp
@@ -63,7 +63,7 @@ DictGui::DictGui(int argc, char **argv, 
 	splash->show_now();			// show splash window NOW
 	splash->flush_queue();		// needed only once
 	
-	history = new History::History(CONF_WORDS_IN_HISTORY, CONF_MAX_WORD_LEN);
+	history = new History(CONF_WORDS_IN_HISTORY, CONF_MAX_WORD_LEN);
 
 	splash->set_step(0.2);
 #endif
--- a/src/dictgui.h
+++ b/src/dictgui.h
@@ -147,8 +147,8 @@ protected:
 
 	Glib::RefPtr<Gtk::TextBuffer> tbuf;
 
-	SigC::Connection con_entry;
-	SigC::Connection con_timer;
+	sigc::connection con_entry;
+	sigc::connection con_timer;
 
 	Glib::RefPtr<Gtk::Clipboard> clp;
 
--- a/src/workhelper.h
+++ b/src/workhelper.h
@@ -59,7 +59,7 @@ private:
 	Gtk::TextView textarea;
 	Gtk::ScrolledWindow scwin;
 	Gtk::VScrollbar *vs;
-	SigC::Connection con_wait;
+	sigc::connection con_wait;
 	
 	unsigned int hide_timeout;
 	unsigned int sizex, sizey;
